containers/layouts/DraftTypeSelect.js

```diff
@@ -1,20 +1,28 @@
 import React from 'react';
 import { c } from 'ttag';
-import { Select } from 'react-components';
+import { Select, useApiWithoutResult, useMailSettings, useEventManager } from 'react-components';
 import { updateDraftType } from 'proton-shared/lib/api/mailSettings';
 
-import useApi from '../../hooks/useApi';
-
 const DraftTypeSelect = () => {
-    const api = useApi();
-    const handleChange = (event) => api(updateDraftType(event.target.value));
+    const { DraftMIMEType } = useMailSettings();
+    const { call } = useEventManager();
+    const { request, loading } = useApiWithoutResult(updateDraftType);
+
+    const handleChange = ({ target }) => async () => {
+        await request(target.value);
+        call();
+    };
 
     const options = [
         { text: c('Option').t`Normal`, value: 'text/html' },
         { text: c('Option').t`Plain Text`, value: 'text/plain' }
     ];
 
-    return <Select options={options} onChange={handleChange} />;
+    return (
+        <>
```

Comment:
Why do you add a fragment here ?

Reply:
I guess the extra <> </> is called a fragment. Actually that is one question I have. When does one have to use it? I used it here because it was used in other LayoutsSection components, but indeed I don't know when to put it

Reply:
You don't need them when you return only one component ex: <Select>.
You need them to prevent an extra div with many elements, ex:

```
<div>
  <ComponentA>....
  <ComponentB>....
</div>
```

vs

```
<>
  <ComponentA>....
  <ComponentB>....
</>
```
cf https://reactjs.org/docs/fragments.html

Reply:
Thanks for the explanation! My bad I haven't reached that section in the React documentation yet